components/Search/Modal/TabSection.tsx

```diff
@@ -0,0 +1,97 @@
+import CheckboxField from '@/components/common/Checkbox';
+import { Tab, Tabs } from '@/components/common/Tab';
+import styled from '@emotion/styled';
+import React from 'react';
+
+type TabSectionProps = {
+  selectedIngredients: string[];
+  setSelectedIngredients: React.Dispatch<React.SetStateAction<string[]>>;
+
+  selectedOnly: boolean;
+
+  categoryIngredientList: CategoryIngredient[];
+};
+
+const TabSection = ({
+  selectedIngredients,
+  setSelectedIngredients,
+  categoryIngredientList,
+  selectedOnly
+}: TabSectionProps) => {
+  // check handle
+  const handleIngredientCheck = (ingredient: string) => {
+    const findIngredient = selectedIngredients.find(
+      (selected) => selected === ingredient
+    );
+
+    if (findIngredient)
+      setSelectedIngredients((prev) =>
+        prev.filter((selected) => selected !== findIngredient)
+      );
+    else setSelectedIngredients((prev) => [...prev, ingredient]);
+  };
+
+  const isChecked = (ingredient: string): boolean =>
+    !!selectedIngredients.find((selected) => selected === ingredient);
+
+  return (
+    <TabsContainer>
+      <Tabs>
+        {categoryIngredientList.map((category: CategoryIngredient) => (
+          <Tab key={category.category_name} title={category.category_name}>
+            <TabWrapper>
+              {category.ingredientNameList.map((ingredient: Ingredient) =>
+                !selectedOnly ? (
+                  <CheckboxFieldWrapper key={ingredient}>
+                    <CheckboxField
+                      name={ingredient}
+                      label={ingredient}
+                      checked={isChecked(ingredient)}
+                      setValue={() => {
+                        handleIngredientCheck(ingredient);
+                      }}
+                    />
+                  </CheckboxFieldWrapper>
+                ) : (
+                  isChecked(ingredient) && (
+                    <CheckboxFieldWrapper key={ingredient}>
+                      <CheckboxField
+                        name={ingredient}
+                        label={ingredient}
+                        checked={isChecked(ingredient)}
+                        setValue={() => {
+                          handleIngredientCheck(ingredient);
+                        }}
+                      />
+                    </CheckboxFieldWrapper>
+                  )
+                )
+              )}
+            </TabWrapper>
+          </Tab>
+        ))}
```

Comment:
```suggestion
        {categoryIngredientList.map((category: CategoryIngredient) => (
          <Tab key={category.category_name} title={category.category_name}>
            <TabWrapper>
              {category.ingredientNameList.map((ingredient: Ingredient) =>
                (!selectedOnly ||  isChecked(ingredient))  && (
                  <CheckboxFieldWrapper key={ingredient}>
                    <CheckboxField
                      name={ingredient}
                      label={ingredient}
                      checked={isChecked(ingredient)}
                      setValue={() => {
                        handleIngredientCheck(ingredient);
                      }}
                    />
                  </CheckboxFieldWrapper>
                ) 
            </TabWrapper>
          </Tab>
        ))}
```

p2) 인라인으로 적은 거라 맞게 했는지 모르겠지만 축약 가능

Reply:
사실 나는 축약을 싫어해... 머리가 빠가라서 내 기준, 한눈에 보기 어려운 거는 최대한 if 로 떨궈버려.. (그 말인 즉슨 사실 ? 도 싫어하는...) 하지만 깔끔한 코드를 위해 꼬우!